### PR TITLE
Fixes/3.7.1 fixes

### PR DESCRIPTION
**Pull request by DRadmir**

### CODE REVIEW

File: fearless/Common/Services/ChainRegistry/ChainRegistry.swift [hunk 1]
```diff
@@ -385,63 +385,26 @@ extension ChainRegistry: ChainRegistryProtocol {
...
         guard
             let chain = chains.first(where: { $0.chainId == chainId }),
-            let currentConnection = getConnection(for: chainId),
-            let currentURL = currentConnection.url
+            let currentConnection = getConnection(for: chainId)
         else {
             return
         }
-        connectionNeedsReconnect(for: chain, previusUrl: currentURL)
+        // TODO: - Remove retry logic if web socket engine refactoring helped
```
Comment (bnsports) [inline fearless/Common/Services/ChainRegistry/ChainRegistry.swift]: still need this?

File: fearless/Common/Services/ChainRegistry/ConnectionPool/ConnectionPool.swift [hunk 2]
```diff
@@ -50,113 +49,67 @@ final class ConnectionPool {
...
-        delegate?.webSocketDidChangeState(url: previousUrl, state: newState)
+        delegate?.webSocketDidChangeState(chainId: chainId, state: newState)
     }
 }
 
-struct DwillerApiKeyInjector {
-    func apiKeyInjectedURL(node: ChainNodeModel) -> URL {
-        guard node.name.lowercased().contains("dwellir") else {
-            return node.url
+struct NodeApiKeyInjector {
```
Comment (bnsports) [inline fearless/Common/Services/ChainRegistry/ConnectionPool/ConnectionPool.swift]: to external file please

File: fearless/Common/Services/ChainRegistry/ConnectionPool/ConnectionPool.swift [hunk 3]
```diff
@@ -50,113 +49,67 @@ final class ConnectionPool {
...
-
-        connectionLock.exclusivelyWrite {
-            self.connectionsByChainIds = self.connectionsByChainIds.filter { $0.key != chainId }
-            self.failedUrls[chainId] = nil
-        }
+    func setDelegate(_ delegate: any ConnectionPoolDelegate) {
+        self.delegate = delegate
     }
+
+    func resetConnection(for _: ChainModel.Id) {}
```
Comment (bnsports) [inline fearless/Common/Services/ChainRegistry/ConnectionPool/ConnectionPool.swift]: empty method

File: fearless/Modules/CrossChain/CrossChainPresenter.swift [hunk 4]
```diff
@@ -113,7 +114,10 @@ final class CrossChainPresenter {
     }
 
     private func checkLoadingState() {
-        view?.setButtonLoadingState(isLoading: !loadingCollector.isReady)
+        guard let isReady = loadingCollector.isReady else {
```
Comment (bnsports) [inline fearless/Modules/CrossChain/CrossChainPresenter.swift:117]: why?

File: fearless/Modules/CrossChain/CrossChainPresenter.swift [hunk 5]
```diff
@@ -263,7 +261,7 @@ final class CrossChainPresenter {
             guard let chain = selectedDestChainModel else {
                 return
             }
-            let isValid = interactor.validate(address: recipientAddress, for: chain).isValid
+            let isValid = interactor.validate(address: recipientAddress, for: chain).isValidOrSame
```
Comment (bnsports) [inline fearless/Modules/CrossChain/CrossChainPresenter.swift:264]: isValidOrSame means that same address is valid. why you need to rename this property? can you just hide 'same-address checking' inside?

File: fearless/Modules/CrossChain/CrossChainViewLayout.swift [hunk 6]
```diff
@@ -93,10 +93,12 @@ final class CrossChainViewLayout: UIView {
     // MARK: - Public methods
 
     func bind(originFeeViewModel: BalanceViewModelProtocol?) {
+        originNetworkFeeView.isHidden = originFeeViewModel == nil
```
Comment (bnsports) [inline fearless/Modules/CrossChain/CrossChainViewLayout.swift:96]: why hidden ? shouldn't be loader there ?

File: fearless/Modules/CrossChain/CrossChainViewLayout.swift [hunk 7]
```diff
@@ -134,6 +141,9 @@ final class CrossChainViewLayout: UIView {
         addSubview(contentView)
         addSubview(actionButton)
 
+        originNetworkFeeView.isHidden = true
```
Comment (bnsports) [inline fearless/Modules/CrossChain/CrossChainViewLayout.swift:144]: why?

File: fearless/Modules/CrossChain/CrossChainViewModelFactory.swift [hunk 8]
```diff
@@ -23,11 +23,11 @@ final class CrossChainViewModelFactory: CrossChainViewModelFactoryProtocol {
         )
     }
 
-    func buildRecipientViewModel(address: String, isValid: Bool) -> RecipientViewModel {
+    func buildRecipientViewModel(address: String) -> RecipientViewModel {
         RecipientViewModel(
             address: address,
             icon: try? iconGenerator.generateFromAddress(address),
-            isValid: isValid,
+            isValid: true,
```
Comment (bnsports) [inline fearless/Modules/CrossChain/CrossChainViewModelFactory.swift:30]: is this property still needed as you just passed hardcoded value?

File: fearless/Modules/NewWallet/WalletTransactionDetails/ViewModel/WalletTransactionDetailsViewModelFactory.swift [hunk 9]
```diff
@@ -36,7 +36,11 @@ class WalletTransactionDetailsViewModelFactory: WalletTransactionDetailsViewMode
             return nil
         }
 
-        let hash = transaction.transactionId
+        var hash = transaction.transactionId
+        if let reefBlockHash = transaction.context?["reefBlockHash"] {
+            let trimmedBlockHash = reefBlockHash.hasPrefix("0x") ? reefBlockHash.dropFirst(2).prefix(5) : reefBlockHash.prefix(5)
```
Comment (bnsports) [inline fearless/Modules/NewWallet/WalletTransactionDetails/ViewModel/WalletTransactionDetailsViewModelFactory.swift:41]: very unsafe. a little change on subsquid side cause a crash on app side

**Review by bnsports — CHANGES_REQUESTED** [review]

File: fearless/Common/Services/ChainRegistry/ChainRegistry.swift (hunk 1 above)
Comment (DRadmir) [reply fearless/Common/Services/ChainRegistry/ChainRegistry.swift]: I added currentConnection.connectIfNeeded() for force connect if possible in engine

File: fearless/Modules/CrossChain/CrossChainPresenter.swift (hunk 4 above)
Comment (DRadmir) [reply fearless/Modules/CrossChain/CrossChainPresenter.swift:117]: cause button should start loading just when ready to load fee

File: fearless/Modules/CrossChain/CrossChainPresenter.swift (hunk 5 above)
Comment (DRadmir) [reply fearless/Modules/CrossChain/CrossChainPresenter.swift:264]: I shouldn't change something inside. Is same address also is valid

File: fearless/Modules/CrossChain/CrossChainViewLayout.swift (hunk 6 above)
Comment (DRadmir) [reply fearless/Modules/CrossChain/CrossChainViewLayout.swift:96]: cause now we doesn't select destination network by default

File: fearless/Modules/CrossChain/CrossChainViewLayout.swift (hunk 7 above)
Comment (DRadmir) [reply fearless/Modules/CrossChain/CrossChainViewLayout.swift:144]: cause now we doesn't select destination network by default

File: fearless/Modules/CrossChain/CrossChainViewModelFactory.swift (hunk 8 above)
Comment (DRadmir) [reply fearless/Modules/CrossChain/CrossChainViewModelFactory.swift:30]: unused in this module, but used in other modules

**Review by bnsports — DISMISSED** [review]

**Review by Viscount88 — DISMISSED** [review]

**Review by Viscount88 — APPROVED** [review]

**Pull request merged**